engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -893,11 +898,100 @@ func (suite *Suite) TestGetSystemTransactionResult_HappyPath() {
 		suite.Require().NoError(err)
 
 		// Expected system chunk transaction
-		systemTx, err := blueprints.SystemChunkTransaction(suite.chainID.Chain())
+		suite.Require().Equal(flow.TransactionStatusExecuted, res.Status)
+		suite.Require().Equal(suite.systemTx.ID(), res.TransactionID)
+
+		// Check for successful decoding of event
+		_, err = jsoncdc.Decode(nil, res.Events[0].Payload)
+		suite.Require().NoError(err)
+
+		events, err = convert.MessagesToEventsWithEncodingConversion(eventMessages,
+			exeNodeEventEncodingVersion,
+			entities.EventEncodingVersion_JSON_CDC_V0)
+		suite.Require().NoError(err)
+		suite.Require().Equal(events, res.Events)
+	})
+}
+
+// TestGetSystemTransactionResultFromCacheWhenFound tests that GetSystemTransactionResult call
+// returns system transaction result for required block id from the cache.
+//
+// How it is tested: In case of cache hit, GetSystemTransactionResult doesn't
+// reach out to the execution nodes and returns early, so we check how
+// many times we do reach execution nodes
+func (suite *Suite) TestGetSystemTransactionResultFromCacheWhenFound() {
```

Comment:
this test should cover looking up the result from storage. it's basically the same test we have for tx results, just using the system tx.